src/runtime-rs/crates/runtimes/common/src/container_manager.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::types::{
+    ContainerConfig, ExecProcessRequest, KillRequest, ProcessExitStatus, ProcessID,
+    ProcessStateInfo, ResizePtyRequest, ShimPid, ShutdownRequest, StatsInfo, UpdateRequest,
+};
+
+#[async_trait]
+pub trait ContainerManager: Send + Sync {
+    // container lifecycle
+    async fn create_container(&self, config: ContainerConfig) -> Result<u32>;
+    async fn pause_container(&self, cid: &str) -> Result<()>;
+    async fn resume_container(&self, cid: &str) -> Result<()>;
+    async fn stats_container(&self, cid: &str) -> Result<StatsInfo>;
+    async fn update_container(&self, req: UpdateRequest) -> Result<()>;
+
+    // process lifecycle
+    async fn close_io_process(&self, process_id: &ProcessID) -> Result<()>;
+    async fn delete_process(&self, process_id: &ProcessID) -> Result<ProcessStateInfo>;
+    async fn exec_process(&self, req: ExecProcessRequest) -> Result<()>;
+    async fn kill_process(&self, req: &KillRequest) -> Result<()>;
+    async fn resize_pty_process(&self, req: &ResizePtyRequest) -> Result<()>;
+    async fn start_process(&self, process_id: &ProcessID) -> Result<u32>;
+    async fn state_process(&self, process_id: &ProcessID) -> Result<ProcessStateInfo>;
+    async fn wait_process(&self, process_id: &ProcessID) -> Result<ProcessExitStatus>;
+
+    // utility
+    async fn pid(&self) -> Result<ShimPid>;
+    async fn connect(&self, _cid: &str) -> Result<ShimPid>;
```

Comment:
Do you want to mark the parameter as unused in the trait?

If all the implementations discard `_cid`, then just remove it. Otherwise, use the `_cid` notation only in the `impl`.

Reply:
Thanks for your review, I think it will be used in the future and renamed to `container_id` with the type `ContainerID`.